PassGen/MainWindow.xaml.cs

```diff
@@ -1,74 +1,74 @@
+        private void GenerateButtonClick(object sender, RoutedEventArgs e)
         {
-            string s = String.Empty;
-            for (int i = 0; i < PLength.Value; i++)
+            // for more info check ASCII table
+
+            var resultPassword = string.Empty;
+
+            const int startShift = ' ' + 1;
+            const int charRange = 'z' - '!' + 1;
+
+            for (var i = 0; i < PasswordLengthSlider.Value; i++)
             {
-                int xchar = 33 + getNewNumber(94);
-                if (Digits.IsChecked == true)
+                var charValue = startShift + GetNextNumber(charRange);
+
+                if (UseDigitsCheckBox.IsChecked == true)
                 {
-                    if (Symbols.IsChecked == false)
+                    if (UseSpecialSymbolsCheckBox.IsChecked == false)
                     {
-                        while (xchar < 48 || xchar > 57 && xchar < 65 || xchar > 90 && xchar < 97 || xchar > 122)
-                        {
-                            xchar = 33 + getNewNumber(94);
-                        }
+                        while (charValue < '0' || charValue > '9' && charValue < 'a' || charValue > 'z' && charValue < 'A' || charValue > 'Z')
+                            charValue = startShift + GetNextNumber(charRange);
                     }
                 }
                 else
                 {
-                    if (Symbols.IsChecked == false)
+                    if (UseSpecialSymbolsCheckBox.IsChecked == false)
                     {
-                        while (xchar < 65 || xchar > 90 && xchar < 97 || xchar > 122)
-                        {
-                            xchar = 33 + getNewNumber(94);
-                        }
+                        while (charValue < 'a' || charValue > 'z' && charValue < 'A' || charValue > 'Z')
+                            charValue = startShift + GetNextNumber(charRange);
                     }
                     else
                     {
-                        while (xchar > 47 && xchar < 58)
-                        {
-                            xchar = 33 + getNewNumber(94);
-                        }
+                        while (charValue > '0' && charValue < '9')
+                            charValue = startShift + GetNextNumber(charRange);
                     }
                 }
-                s += Convert.ToChar(xchar);
-            };
-            Result.Text = s;
+
+                resultPassword += Convert.ToChar(charValue);
+            }
+            ResultTextBox.Text = resultPassword;
         }
 
-        private void PLength_ValueChanged(object sender, RoutedPropertyChangedEventArgs<double> e)
+        private void PasswordLengthSliderValueChanged(object sender, RoutedPropertyChangedEventArgs<double> e)
         {
-            Current.Content = PLength.Value;
+            if (CurrentPasswordLengthLabel != null)
+                CurrentPasswordLengthLabel.Content = PasswordLengthSlider.Value;

```

Comment:
Why not use `variable?` syntax for checking for null?

Reply:
How can you use it here? Provide example :)

Reply:
```suggestion
                CurrentPasswordLengthLabel?.Content = PasswordLengthSlider.Value;
```

Reply:
![](http://1488.me/radioegor146/I3CkU7C1zl.png)
:thinking:

Reply:
Well, this is weird